store/rootmulti/store.go

```diff
@@ -66,25 +66,53 @@ func NewStore(dir string, logger log.Logger, sdk46Compact bool) *Store {
 	}
 }
 
-// Implements interface Committer
-func (rs *Store) Commit() types.CommitID {
+// flush writes all the pending change sets to memiavl tree.
+func (rs *Store) flush() error {
 	var changeSets []*memiavl.NamedChangeSet
 	for key := range rs.stores {
 		// it'll unwrap the inter-block cache
 		store := rs.GetCommitKVStore(key)
 		if memiavlStore, ok := store.(*memiavlstore.Store); ok {
-			changeSets = append(changeSets, &memiavl.NamedChangeSet{
-				Name:      key.Name(),
-				Changeset: memiavlStore.PopChangeSet(),
-			})
-		} else {
-			_ = store.Commit()
+			cs := memiavlStore.PopChangeSet()
+			if len(cs.Pairs) > 0 {
+				changeSets = append(changeSets, &memiavl.NamedChangeSet{
+					Name:      key.Name(),
+					Changeset: cs,
+				})
+			}
 		}
 	}
 	sort.SliceStable(changeSets, func(i, j int) bool {
 		return changeSets[i].Name < changeSets[j].Name
 	})
-	_, _, err := rs.db.Commit(changeSets)
+
+	return rs.db.ApplyChangeSets(changeSets)
+}
+
+// WorkingHash returns the app hash of the working tree,
+// it's only compatible with sdk 0.47 and later.
+//
+// Implements interface Committer.
+func (rs *Store) WorkingHash() []byte {
```

Comment:
Don't we also need to modify GetWorkingHash?

Reply:
In sdk 0.50, it's named as WorkingHash, should be the same thing as the GetWorkingHash in sei i think